testing/helpers/trie.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package helpers
+
+import (
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/utils"
+)
+
+type LinearCongruentialGenerator struct {
+	seed uint64
+}
+
+func NewGenerator() *LinearCongruentialGenerator {
+	return &LinearCongruentialGenerator{}
+}
+
+func (rng *LinearCongruentialGenerator) Next() uint16 {
+	rng.seed = (rng.seed*1140671485 + 12820163) % 65536
```

Comment:
This is because when Flow implemented their trie, they followed a specification which used this specific PRNG for its tests, so that they could exactly replicate the tests of this original spec. I chose to do the same, but when we're allowed to do breaking changes, we'll be able to get rid of it.